csharp/autobuilder/Semmle.Autobuild/DotNetRule.cs

```diff
@@ -56,13 +56,13 @@ public BuildScript Analyse(Autobuilder builder, bool auto)
                 });
         }
 
-        static BuildScript WithDotNet(Autobuilder builder, Func<string, IDictionary<string, string>, bool, BuildScript> f)
+        static BuildScript WithDotNet(Autobuilder builder, Func<string?, IDictionary<string, string>?, bool, BuildScript> f)
         {
-            var installDir = builder.Actions.PathCombine(builder.Options.RootDirectory, ".dotnet");
+            string? installDir = builder.Actions.PathCombine(builder.Options.RootDirectory, ".dotnet");
```

Comment:
Why nullable? `PathCombine` does not return `null`.

Reply:
`installDir` is set to `null` on line 79. I'm trying to avoid rewriting the logic too much.